### ISSUE TITLE
Problem to close a Window with a focused TextField

### ISSUE THREAD

**Issue by mlopezFC** [vaadin/framework#10642]
This was reproduced in Vaadin 8.2.0 y 8.3.1
If I create a Window with a TextField on it, if the TextField has focus, it's necessary to press Escape key twice to close the window, but without the focusing, pressing the Escape key only once closes it.
It's easy to reproduce it, just create a minimal Vaadin project with this UI:

```
@Theme("mytheme")
public class MyUI extends UI {

	@SuppressWarnings("serial")
	@Override
	protected void init(VaadinRequest vaadinRequest) {
		final VerticalLayout layout = new VerticalLayout();

		Button button1 = new Button("open window");
		button1.addClickListener(event -> {

			Window window = new Window();
			window.setModal(true);

			TextField textField = new TextField("focus me");
			window.setContent(textField);

			UI.getCurrent().addWindow(window);
			// uncomment this line to reproduce the problem:
			//textField.focus();

		});

		layout.addComponents(button1);

		setContent(layout);
	}

	@WebServlet(urlPatterns = "/*", name = "MyUIServlet", asyncSupported = true)
	@VaadinServletConfiguration(ui = MyUI.class, productionMode = false)
	public static class MyUIServlet extends VaadinServlet {
	}
}
```
If you test it, after pressing the button, a Window will popup, if you press Escape once, the window will close. If you uncomment the line `textField.focus();` and try again, you will see that you have to press Escape key twice to close the window.
Regards.

**Comment by onlineandy** [comment]
I can reproduce it with 8.5.1.

**Comment by ZheSun88** [comment]
Bug has been fixed with #11265
